File: 1_twoSum/twoSum.ts
function soma(nums: Array<number>, target: number) {
  for (let index = 0; index < nums.length; index++) {
    for (let index2 = index + 1; index2 < nums.length; index2++) {
      if (nums[index] + nums[index2] === target) {
        return [index, index2];
      }
    }
  }
}

let nums = [2, 11, 15, 7];
let target = 9;
console.log(soma(nums, target));

File: 1_twoSum/twoSum.go
package main

import "fmt"

var alf string

func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for i, num := range nums {
		complement := target - num
		if j, exists := m[complement]; exists {
			return []int{j, i}
		}
		m[num] = i
	}
	return nil
}
func main() {
	nums := []int{3, 2, 4}
	fmt.Println(twoSum(nums, 6))
}
